src/SuperVault.sol

```diff
@@ -463,23 +487,48 @@ contract SuperVault is BaseStrategy, ISuperVault {
 
         bytes[] memory dataToEncode = new bytes[](_numberOfSuperforms_);
 
-        for (uint256 i; i < _numberOfSuperforms_; ++i) {
+        uint256[] memory _weights_ = weights;
+        for (uint256 i; i < _numberOfSuperforms_;) {
             mvData.liqRequests[i].token = address(asset);
 
             (address superform,,) = mvData.superformIds[i].getSuperform();
 
             dataToEncode[i] = _prepareDepositExtraFormDataForSuperform(mvData.superformIds[i]);
 
             /// @notice rounding down to avoid one-off issue
-            mvData.amounts[i] = amount_.mulDiv(weights[i], TOTAL_WEIGHT, Math.Rounding.Down);
+            mvData.amounts[i] = amount_.mulDiv(_weights_[i], TOTAL_WEIGHT, Math.Rounding.Down);
             mvData.outputAmounts[i] = IBaseForm(superform).previewDepositTo(mvData.amounts[i]);
             mvData.maxSlippages[i] = MAX_SLIPPAGE;
+
+            unchecked {
+                ++i;
+            }
         }
 
         mvData.extraFormData = abi.encode(_numberOfSuperforms_, dataToEncode);
         return mvData;
     }
 
+    function _prepareSingleVaultDepositData(uint256 amount_) internal view returns (SingleVaultSFData memory svData) {
+        svData.superformId = superformIds[0];
+        svData.amount = amount_;
+        svData.maxSlippage = MAX_SLIPPAGE;
+        svData.liqRequest.token = address(asset);
+        svData.hasDstSwap = false;
+        svData.retain4626 = false;
+        svData.receiverAddress = address(this);
+        svData.receiverAddressSP = address(this);
+
+        (address superform,,) = svData.superformId.getSuperform();
+        svData.outputAmount = IBaseForm(superform).previewDepositTo(amount_);
+        bytes memory dataToEncode = _prepareDepositExtraFormDataForSuperform(svData.superformId);
+        bytes[] memory finalDataToEncode = new bytes[](1);
+        finalDataToEncode[0] = dataToEncode;
+        svData.extraFormData = abi.encode(1, finalDataToEncode);
```

Comment:
Doublecheck: doesn't this have to be an encoded array of 1 element?
Otherwise the 1 should be implicit and maybe can be removed